sftp/src/main/scala/blobstore/sftp/SftpStore.scala

```diff
@@ -48,20 +49,18 @@ final class SftpStore[F[_]](
       ch
     }
     semaphore.fold(openF){s =>
-      F.ifM(s.tryAcquire)(openF, getChannel)
+      s.tryAcquire.ifM(openF, getChannel)
     }
   }
 
-  private val getChannel = F.flatMap(mVar.tryTake) {
-    case Some(channel) => F.pure(channel)
+  private val getChannel = mVar.tryTake.flatMap {
+    case Some(channel) => channel.pure[F]
     case None => openChannel
   }
 
-  private def channelResource: Resource[F, ChannelSftp] = Resource.make{
-    getChannel
-  }{
-    case ch if ch.isClosed => F.unit
-    case ch => F.ifM(mVar.tryPut(ch))(F.unit, SftpStore.closeChannel(semaphore, blocker)(ch))
+  private def channelResource: Resource[F, ChannelSftp] = Resource.make(getChannel) {
+    case ch if ch.isClosed => Applicative[F].unit
```

Comment:
Consider `Applicative[F].unit` -> `().pure[F]` in vein of using syntax.
Two other occurrences in this file.